File: 2015/day02/day02part2.rb
#!/usr/bin/env ruby

paper = 0
ribbon = 0
ARGF.each do |line|
    l,w,h = line.split('x').map{|x| x.to_i}
    paper += 2*l*w + 2*w*h + 2*h*l + [l*w, w*h, h*l].min
    [l,w,h].min(2).each do |x|
        ribbon += 2*x
    end
    ribbon += l*w*h
end
puts "#{paper} square feet of paper"
puts "#{ribbon} feet of ribbon"

File: 2015/day02/day02part2.py
#!/usr/bin/env python3
import sys
infile = sys.argv[0].split('part')[0] + '.input'


total_ribbon = 0

with open(infile, 'r') as input:
    for line in input:
        l,w,h = map(int, line.split('x'))
        dim = [l,w,h]
        dim.sort()
        dim.pop()
        x,y = dim
        total_ribbon += x+x+y+y + l*w*h

print(f'{total_ribbon} feet of ribbon paper')
